rules/aip0134/message_rules.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0134
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+func extractResource(reqName string) string {
+	// Strips "Update" from the beginning and "Request" from the end.
+	return reqName[6:len(reqName)-7]
+}
+
+// The Update standard method should only have expected fields.
+var standardFields = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0134", "request-message", "name-field"),
+	URI:    "https://aip.dev/134#request-message",
+	OnlyIf: isUpdateRequestMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		r := extractResource(m.GetName())
+		fields := []struct {
+			name string
+			typ  string
+		}{
+			{strings.ToLower(r), r},
```

Comment:
You need `strings.ToLower(strcase.SnakeCase(r))`.

Otherwise, a resource `FooBar` will ask for field name `foobar` (wrong) rather than `foo_bar` (right).

Reply:
Very good point, done (and updated tests).